src/pylhe/__init__.py

```diff
@@ -323,6 +323,10 @@ def read_lhe_with_attributes(filepath):
                         else:
                             eventdict["optional"].append(p.strip())
                     for sub in element:
+                        if sub.tag == "weights":
+                            for i, w in enumerate(sub.text.split()):
+                                if w:
+                                    eventdict["weights"][i] = float(w)
```

Comment:
One thing I'd like to improve here is to use the label provided in the `initrwgt` block for given index of a weight. But that does not get loaded by default right. Should the `lhe_init` be called here too to translate from index to label.

Reply:
eg. for this file it has this sort of init

```
<initrwgt>
<weightgroup name='nominal' combine='None' >
<weight id='MUR1.0_MUF1.0_PDF99208082' >default</weight>
</weightgroup>
<weightgroup name='scale_variations' combine='None' >
<weight id='MUR2.0_MUF2.0_PDF99208082' > renscfact=2d0 facscfact=2d0 </weight>
```
so instead of `weights = {0: XYZ,...}` I'd prefer `weights={'MUR2.0_MUF2.0_PDF99208082': XYZ,...}`
What do you think?